src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-realm.service.ts

```diff
@@ -23,6 +23,17 @@ export class RgwRealmService {
     });
   }
 
+  update(realm: RgwRealm, defaultRealm: boolean, newRealmName: string) {
+    return this.rgwDaemonService.request((requestBody: any) => {
+      requestBody = {
+        realm_name: realm.name,
+        default: defaultRealm,
+        new_realm_name: newRealmName
+      };
+      return this.http.put(`${this.url}/${realm.name}`, requestBody);
+    });
+  }
```

Comment:
Do we really need to pass an instance of `RgwRealm`? Only the name is being used.
A simple string will do the job.

Reply:
Since we have modal created for rgw-realm, so it doesn't make sense to pass individual strings as realm name, we should use the object from realm class, otherwise there's no point to create the modal at the first place.